routing-daemon/lib/openshift/routing/models/f5-icontrol-rest.rb

```diff
@@ -235,11 +185,84 @@ def delete_pool_alias pool_name, alias_str
+                 "command" => "install",
+                 "name" => "#{alias_str}-https-cert",
+                 "from-local-file" => "/var/tmp/#{URI.escape(alias_str)}.crt"
+               }.to_json)
+
+        @logger.debug("LTM cert to be installed /var/tmp/#{URI.escape(alias_str)}.key")
+        post(url: "https://#{@host}/mgmt/tm/sys/crypto/key",
+               payload: {
+                 "command" => "install",
+                 "name" => "#{alias_str}-https-key",
+                 "from-local-file" => "/var/tmp/#{URI.escape(alias_str)}.key"
+               }.to_json)
+
+        @logger.debug("LTM creating client-ssl profile with #{URI.escape(alias_str)}.crt and #{URI.escape(alias_str)}.key")
+        post(url: "https://#{@host}/mgmt/tm/ltm/profile/client-ssl",
+                 payload: {
+                   "name" => "#{URI.escape(alias_str)}-ssl-profile",
+                   "cert" => "#{URI.escape(alias_str)}-https-cert.crt",
+                   "key" => "#{URI.escape(alias_str)}-https-key.key",
+                   "serverName" => "#{URI.escape(alias_str)}"
+                 }.to_json)
+
+        @logger.debug("LTM adding #{URI.escape(alias_str)}-ssl-profile client-ssl to #{@https_vserver}")
+        post(url: "https://#{@host}/mgmt/tm/ltm/virtual/#{@https_vserver}/profiles",
+                 payload: {
+                   "name" => "#{URI.escape(alias_str)}-ssl-profile",
+                   "context" => "clientside",
+                 }.to_json)
+
+        # Requires LTM System->Users->admin terminal setting to be set to advanced (bash)
+        @logger.debug("LTM removing temporary. rm -f /var/tmp/#{alias_str}.crt")
+        result = `ssh -i #{@ssh_private_key} admin@#{@host} 'rm -f /var/tmp/alias_str}.crt'`
+        @logger.debug("LTM removing temporary. rm -f var/tmp/#{alias_str}.key")
+        result = `ssh -i #{@ssh_private_key} admin@#{@host} 'rm -f /var/tmp/#{alias_str}.key'`
+      rescue Errno::ENOENT
+        # Nothing to do;
+      ensure
+        File.unlink(certfname)
+        File.unlink(keyfname)
+      end
+    end
 
     def remove_ssl pool_name, alias_str
-      # no-op
+      @logger.debug("LTM removing #{URI.escape(alias_str)}-ssl-profile client-ssl from #{@https_vserver}")
+      delete(url: "https://#{@host}/mgmt/tm/ltm/virtual/#{@https_vserver}/profiles/#{URI.escape(alias_str)}-ssl-profile")
+
+      @logger.debug("LTM deleting removing #{URI.escape(alias_str)}-ssl-profile")
+      delete(url: "https://#{@host}/mgmt/tm/ltm/profile/client-ssl/#{URI.escape(alias_str)}-ssl-profile")
+
+      @logger.debug("LTM removing #{alias_str}-https-key")
+      delete(url: "https://#{@host}/mgmt/tm/sys/file/ssl-key/#{URI.escape(alias_str)}-https-key.key")
+
+      @logger.debug("LTM removing #{alias_str}-https-cert")
+      delete(url: "https://#{@host}/mgmt/tm/sys/file/ssl-cert/#{URI.escape(alias_str)}-https-cert.crt")
+    end
+
+    def get_pool_certificates pool_name
+      pool_certs = []
+      return pool_certs
```

Comment:
Yeah, the simple controller calls `get_pool_certificates` from `get_certificates`, but nothing seems to call `get_certificates`.  Can we delete it? Or maybe we should expose the list of certificates through `oo-admin-ctl-routing`?